socorro/external/crashstorage_base.py

```diff
@@ -1266,3 +1282,76 @@ def remove(self, crash_id):
             self.tag,
             end_time - start_time
         )
+
+
+class MetricsEnabledBase(RequiredConfig):
+    """Base class for capturing metrics for crashstorage classes"""
+    required_config = Namespace()
+    required_config.add_option(
+        'metrics_prefix',
+        doc='a string to be used as the prefix for metrics keys',
+        default=''
+    )
+    required_config.add_option(
+        'active_list',
+        default='save_raw_and_processed,act',
+        doc='a comma delimeted list of counters that are enabled',
+        from_string_converter=str_to_list
+    )
+
+    def __init__(self, config, namespace='', quit_check_callback=None):
+        self.config = config
+        self.metrics = markus.get_metrics(self.config.metrics_prefix)
+
+    def _make_key(self, *args):
+        parts = list(args)
+        return '.'.join(x for x in parts if x)
+
+
+class MetricsCounter(MetricsEnabledBase):
+    """Counts the number of times it's called"""
+    required_config = Namespace()
+
+    def __getattr__(self, attr):
+        if attr in self.config.active_list:
+            self.metrics.incr(attr)
+        return self._noop
```

Comment:
This class is similar to `StatsdCounter`, but that one has a bug where it counts *all* calls--not just the ones in the active list.

https://github.com/mozilla-services/socorro/blob/922154e09677bfeb4e07828c9fee305c2cb50b81/socorro/external/statsd/statsd_base.py#L91